### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by kalebdf** [checkr/flagr#100]
Hi @zhouzhuojie,
This is a fantastic project! Do you & the Checkr team have a roadmap or vision for this project?

In no particular order, here are some features I've used in similar projects:
- GET All Flags - Limit, Pagination, Filter (by active/inactive, by tag, by namespace, etc)
- Tagging of Flags (for filtering sake, different apps)
- Namespaces of Flags ([Planout](https://facebook.github.io/planout/docs/namespaces.html))
- Record forcing user into variation bucket (future calls requesting variation for user will return forced variation)

**Comment by zhouzhuojie** [comment]
Thank you! These are awesome feature requests! We have plans on the data metrics side, and if it's necessary, I can work on the features like search flags, flag tags during our next hack days. 

Namespace (i.e. layers) is awesome! I've seen implementation before in other systems, and typically it's not needed all the time, I will probably defer that to future major releases.

Record forcing users into variation bucket or Sticky rollout looks good, but I don't have a clear spec in mind. Right now flagr uses EntityID to hash users into variation bucket. If you don't change the rollout percent or distribution, the same user should always be bucketed into the same variant.

PRs are always welcome! 🎉

**Comment by kalebdf** [comment]
@zhouzhuojie Stellar! We currently are sending all of our `Experiment Viewed` metrics into Segment which flows into our data warehouse / data lake. 🏢  📊 

I looked into `go-swagger` briefly to try to add filtering, limiting, & pagination, but didn't find it. I'm sure they have it though.

The use case for recording user (ID: 123) forced into variation bucket (ID: XYZ) and then having that stick is for the fallback case. Upon service timeout, we can default the feature experience for the user to variation (ID: XYZ). It's not high up on my list at all. I only noticed it on SeatGeek's [Sixpack](http://sixpack.seatgeek.com).

If we build upon **Flagr**, we'll definitely submit a PR. Thanks! 👍
